### PR TITLE
"Money in the XRP Ledger" intro and related

### PR DESCRIPTION
**Pull request by mDuo13**
Display versions (VPN required):
- [Money in the XRP Ledger](https://clients-staging-private.ripple.com/docs/other/staged-dev-portal-docs/xrp-money-intro/concept-money.html) (intro/concept)
- [Currency Format Reference](https://clients-staging-private.ripple.com/docs/other/staged-dev-portal-docs/xrp-money-intro/reference-currency.html)
- [Demurrage](https://clients-staging-private.ripple.com/docs/other/staged-dev-portal-docs/xrp-money-intro/concept-demurrage.html) (for archival purposes, basically)

I got kind of carried away with the "intro" page and ended up needing to split some of the content off into new pages. On the plus side, this also ends up migrating the https://wiki.ripple.com/Currency_format page.

And, because I didn't want to remove one reference to the wiki while adding another, I also migrated https://wiki.ripple.com/Gateway_demurrage . But seriously, reviewers, you can mostly ignore the demurrage page.

(DOC-1088, DOC-229)

### CODE REVIEW

File: content/concept-money.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+Some advanced features of the XRP Ledger, such as [Escrow](concept-escrow.html) and [Payment Channels](tutorial-paychan.html), only work with XRP. Order book [autobridging](https://ripple.com/dev-blog/introducing-offer-autobridging/) uses XRP to deepen liquidity in the decentralized exchange by merging order books of two issued currencies (such as USD:EUR) with XRP order books (USD:XRP + XRP:EUR results in more trades for the USD:EUR book).
+
+XRP also serves as a protective measure against spamming the network. All XRP Ledger addresses need a small amount of XRP to pay the costs of maintaining the XRP Ledger. The [transaction cost](concept-transaction-cost.html) and [reserve](concept-reserves.html) are neutral fees denoted in XRP and not paid to any party.
```
Comment (nbougalis) [inline content/concept-money.md]: Is **denominated** a better word than **denoted** in: 

>The transaction cost and reserve are neutral fees denoted in XRP and not paid to any party

File: content/concept-money.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+For more information on XRP's use cases, benefits, and news, see the [XRP Portal](https://ripple.com/xrp-portal/).
+
+### XRP Properties
+
+XRP is the only native currency in the XRP Ledger. The very first ledger contained 100 billion XRP, and no new XRP can be created. XRP can be destroyed by [transaction costs](concept-transaction-cost.html) or lost by sending it to addresses for which no one holds a key, so XRP is slightly [deflationary](https://en.wikipedia.org/wiki/Deflation) by nature. No need to worry about running out, though: at the current rate of destruction, it would take at least 70,000 years to destroy all XRP, and XRP prices and fees can be adjusted as the total supply of XRP changes.
```
Comment (nbougalis) [inline content/concept-money.md]: Do we want to link "addresses for which no one holds a key" to a page about accounts for which no one holds a key (ACCOUNT_ZERO and ACCOUNT_ONE, and some of the known blackholes?)

File: content/concept-money.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+Issued currencies can be traded with XRP or each other in the XRP Ledger's decentralized exchange.
+
+In the typical model, an issued currency is tied to holdings of currency or other assets outside the XRP Ledger. The issuer of the currency, called a _gateway_, handles deposits and withdrawals to exchange currency outside the XRP Ledger for equivalent balances of issued currency in the XRP Ledger. For more information on how to run a gateway, see the [Gateway Guide](tutorial-gateway-guide.html).
+
+There are other use cases for issued currencies in the XRP Ledger. For example, you can create an "Initial Coin Offering" (ICO) by issuing a fixed amount of currency to a secondary address, then "throwing away the key" to the issuer. (Be aware: ICOs may be [regulated as securities](https://www.sec.gov/oiea/investor-alerts-and-bulletins/ib_coinofferings) in the USA.) Ripple recommends researching the relevant regulations before engaging in any financial service business.
```
Comment (nbougalis) [inline content/concept-money.md]: I'm not sure if our current layout engine allows this, but can we have the "Be aware" section be surrouded by its own frame, and carry a warning icon?

**Review by nbougalis — APPROVED** [review]

File: content/reference-currency.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+## String Formatting
+
+XRP Ledger APIs generally use strings, rather than native JSON numbers, to represent numeric amounts of currency for both XRP and issued currencies. This protects against a loss of precision when using JSON parsers, which may automatically try to represent all JSON numbers in a floating-point format.
+
+## XRP Precision
+
+XRP has the same precision as a 64-bit unsigned integer where each unit is equivalent to 0.000001 XRP. Its properties are:
+
+* Minimum value: `0`
+* Maximum value: `100000000000` (10^11) XRP
```
Comment (nbougalis) [inline content/reference-currency.md]: Can you insert arbitrary HTML? If so, maybe `10<sup>11</sup>` would look better.

File: content/concept-money.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
+# Money in the XRP Ledger
+
+The XRP Ledger is an advanced blockchain-like system that was designed to let people transact in multiple currencies. Users of the XRP Ledger can seamlessly track, trade, and settle multiple currencies in an exchange that's as decentralized as the network itself. Users can issue their own currency-like digital assets, which may or may not represent obligations owed outside the XRP Ledger. Tying it all together is XRP, the XRP Ledger's native cryptocurrency, which serves anti-spam purposes and acts as a medium of exchange.
```
Comment (bachase) [inline content/concept-money.md]: Consider flipping `anti-spam purposes` and `acts as a medium of exchange`. The current order might give the impression that anti-spam is the more important use case.

File: content/concept-demurrage.md [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+
+- Taking display values at a given time and converting them to ledger values to be recorded.
+- Taking ledger values and converting them to a display value at a given point in time.
+
+### Calculating Demurrage
+
+The full formula for calculating demurrage on a currency is as follows:
+
+```
+D = A × ( e ^ (t ÷ τ) )
```
Comment (bachase) [inline content/concept-demurrage.md:23]: Not for this PR, and especially not for a deprecated feature documentation, consider adding [katex](https://github.com/Khan/KaTeX) or [mathjax](https://www.mathjax.org/) for math typesetting

**Review by bachase — APPROVED** [review]
Nice additions!

File: content/concept-demurrage.md [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+4. Take the number of seconds in one year (31536000) and divide by the natural log result from the previous step. For example, **31536000 ÷ -0.005012541823544286 = -6291418827.045599**. This result is the e-folding time in seconds.
+
+**Note:** By convention, Ripple's interest/demurrage rules use a fixed number of seconds per year (31536000), which is not adjusted for leap days or leap seconds.
+
+## Client Support
+
+To support interest-bearing and demurraging currencies, client applications must implement several features:
+
+- When displaying the amount of a demurraging currency retrieved from ledger or transaction data, the client must convert from the ledger value to the display value. (With demurrage, the display values are smaller than the ledger values.)
+- When accepting input for a demurraging currency, the client must convert amounts from a display format to the ledger format. (With demurrage, the ledger values are are larger than the user input value.) The client must use the ledger value when creating payments, offers, and other types of transaction.
```
Comment (ryangyoung) [inline content/concept-demurrage.md]: To improve readability a bit, consider adding a line between these bullet points.

File: content/concept-demurrage.md [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+- Clients must distinguish between currencies that do and do not have interest or demurrage, and among currencies that have different rates of interest or demurrage. Clients should be able to parse the [Interest-Bearing Currency Code Format](#interest-bearing-currency-code-format) into a display such as "XAU (-0.5% pa)".
+
+### ripple-lib Support
+
+Demurrage was supported in ripple-lib versions **0.7.37** through **0.12.9**. Demurrage is ***not supported*** in [RippleAPI](reference-rippleapi.html).
```
Comment (ryangyoung) [inline content/concept-demurrage.md:79]: super nit: Is that the right formatting for ripple-lib?

File: content/concept-money.md [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+
+XRP also serves as a protective measure against spamming the network. All XRP Ledger addresses need a small amount of XRP to pay the costs of maintaining the XRP Ledger. The [transaction cost](concept-transaction-cost.html) and [reserve](concept-reserves.html) are neutral fees denominated in XRP and not paid to any party. In the ledger's data format, XRP is stored in [AccountRoot objects](reference-ledger-format.html#accountroot).
+
+For more information on XRP's use cases, benefits, and news, see the [XRP Portal](https://ripple.com/xrp-portal/).
+
+### XRP Properties
+
+The very first ledger contained 100 billion XRP, and no new XRP can be created. XRP can be destroyed by [transaction costs](concept-transaction-cost.html) or lost by sending it to addresses for which no one holds a key, so XRP is slightly [deflationary](https://en.wikipedia.org/wiki/Deflation) by nature. No need to worry about running out, though: at the current rate of destruction, it would take at least 70,000 years to destroy all XRP, and XRP [prices and fees can be adjusted](concept-fee-voting.html) as the total supply of XRP changes.
```
Comment (ryangyoung) [inline content/concept-money.md:17]: I know this a limitation of my own understanding, but how does the "total supply" change if the "first ledger contained 100 billion XRP, and no new XRP can be created"?

File: content/concept-money.md [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+
+In technical contexts, XRP is measured precisely to the nearest 0.000001 XRP, called a "drop" of XRP. The [`rippled` APIs](reference-rippled.html) require all XRP amounts to be specified in drops of XRP. For example, 1 XRP is represented as `1000000` drops. For more detailed information, see the [currency format reference](reference-currency.html).
+
+## Issued Currencies
+
+All currencies other than XRP are represented as **issued currencies**. These digital assets, sometimes called "issuances" or "IOUs", are tracked in accounting relationships, called "trust lines," between addresses. Issued currencies are typically considered as liabilities from one perspective and assets from the other, so the balance of a trust line is negative or positive depending on which side you view it from. Any address may freely issue (non-XRP) currencies, limited only by how much other addresses are willing to hold.
```
Comment (ryangyoung) [inline content/concept-money.md]: > sometimes called "issuances" or "IOUs"

Maybe de-emphasize this a bit (to reduce naming confusion)  by placing it in parens?

> All currencies other than XRP are represented as **issued currencies**. These digital assets (sometimes also called "issuances" or "IOUs") are tracked...

We should take another shot at creating a Ripple glossary (out of scope for this PR, obviously).

**Review by jbheron — APPROVED** [review]
Great additions! Initially, I had wanted to make a bunch of style suggestions, but sat on them. In reviewing again, I like the language approach you've applied. 

The Money in the XRP Ledger page is a great onboarding document to the XRP ecosystem.

LGTM

File: content/concept-money.md [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+
+Issued currencies can be traded with XRP or each other in the XRP Ledger's decentralized exchange.
+
+In the typical model, an issued currency is tied to holdings of currency or other assets outside the XRP Ledger. The issuer of the currency, called a _gateway_, handles deposits and withdrawals to exchange currency outside the XRP Ledger for equivalent balances of issued currency in the XRP Ledger. For more information on how to run a gateway, see the [Gateway Guide](tutorial-gateway-guide.html).
+
+There are other use cases for issued currencies in the XRP Ledger. For example, you can create an "Initial Coin Offering" (ICO) by issuing a fixed amount of currency to a secondary address, then "throwing away the key" to the issuer.
+
+**Warning:** ICOs may be [regulated as securities](https://www.sec.gov/oiea/investor-alerts-and-bulletins/ib_coinofferings) in the USA.
+
+Ripple recommends researching the relevant regulations before engaging in any financial service business.
```
Comment (ryangyoung) [inline content/concept-money.md]: I feel like this should be stronger somehow. I don't have a good suggestion for how to do that, though, so this is probably acceptable.

File: content/concept-money.md [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+
+Some advanced features of the XRP Ledger, such as [Escrow](concept-escrow.html) and [Payment Channels](tutorial-paychan.html), only work with XRP. Order book [autobridging](https://ripple.com/dev-blog/introducing-offer-autobridging/) uses XRP to deepen liquidity in the decentralized exchange by merging order books of two issued currencies with XRP order books to create synthetic combined order books. (For example, autobridging matches USD:XRP and XRP:EUR orders to augment USD:EUR order books.)
+
+XRP also serves as a protective measure against spamming the network. All XRP Ledger addresses need a small amount of XRP to pay the costs of maintaining the XRP Ledger. The [transaction cost](concept-transaction-cost.html) and [reserve](concept-reserves.html) are neutral fees denominated in XRP and not paid to any party. In the ledger's data format, XRP is stored in [AccountRoot objects](reference-ledger-format.html#accountroot).
+
+For more information on XRP's use cases, benefits, and news, see the [XRP Portal](https://ripple.com/xrp-portal/).
+
+### XRP Properties
```
Comment (ryangyoung) [inline content/concept-money.md:15]: A little to the side of this topic, but is it worth mentioning Escrow in this section?

File: content/reference-currency.md [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+
+The internal format consists of three parts: a sign bit, a mantissa, and an exponent. The mantissa contains up to 15 significant digits (in base-10), and the exponent contains a scale (from -96 to +80). The sign bit indicates whether the amount is positive or negative. Before recording any amount, `rippled` "canonicalizes" the value so that the mantissa is within the range `1000000000000000` to `9999999999999999` (inclusive) and the exponent is in the range -96 to 80 (inclusive). (Exception: the value `0` is represented with a mantissa of `0`.) For example, the canonical representation of 1 unit of currency is `1000000000000000e-15`. The internal calculations generally use integer math so that numbers are always precise within 15 digits.
```
Comment (ryangyoung) [inline content/reference-currency.md]: _comment from ignorance_ Anyone reading this page would know what a mantissa is, right? (I had to look it up.)

**Review by ryangyoung — APPROVED** [review]
Great article, @mDuo13. I left a few very minor, non-substantive comments, but definitely nothing that should block this from being merged.

File: content/concept-demurrage.md (hunk 8 above)
Comment (mDuo13) [reply content/concept-demurrage.md:79]: The "right" formatting for ripple-lib is undefined. This seems to be one of the most common ways to write it, so I'm going to stick with that.

File: content/concept-money.md (hunk 9 above)
Comment (mDuo13) [reply content/concept-money.md:17]: It changes by decreasing as XRP is destroyed, as described in this paragraph. Maybe if I said "available supply" instead that would make more sense?

File: content/concept-money.md (hunk 11 above)
Comment (mDuo13) [reply content/concept-money.md]: how about "strongly recommends"? ;)

File: content/reference-currency.md (hunk 13 above)
Comment (mDuo13) [reply content/reference-currency.md]: Not necessarily, unfortunately. It's something I had to look up, too. Sadly, the definitions I found were in some ways more confusing than ever.

Looking it over again, it's possible to rephrase the entire section to use "significant digits" exclusively and avoid the word "mantissa", which is probably more accessible to more readers, so I'm going to do that.

**Pull request merged**